pkg/allocation/converters/converter_test.go

```diff
@@ -1533,3 +1543,51 @@ func TestConvertAllocationResponseToGSA(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertStateV1ToError(t *testing.T) {
```

Comment:
Oh, I just saw this. Hmm. Honestly if you make the changes above, I don't think you really need this test - the errors are already checked above, and you can add a simple permutation of grpcUnallocatedStatusCode to check your new logic.